pacu/modules/organizations__assume_role/main.py

```diff
@@ -0,0 +1,150 @@
+parser.add_argument('--accounts-file', required=False, nargs=1, default=None, help="Pass in a filename containing a list of accounts to try assuming")
+parser.add_argument('--roles', required = False, nargs = '+', default=None,  help='Pass in a list of roles to try assuming. If non are specified uses default OrganizationAccountAccessRole, --roles a b c')
+parser.add_argument('--roles-file', required=False, nargs=1, default=None, help='Pass in a filename containing a list of roles to try assuming')
+#parser.add_argument('--create-sessions', required=False, action="store_true", help='Will automatically create a session for every successful AssumeRole operation. Can check all of them after with swap_keys command.')
+
+roles_list = []
+
+def assume_role(args, session, role_arn):
+    global roles_list
+    sts = boto3.client('sts')
+    try:
+        resp = sts.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName=''.join(random.choice(string.ascii_letters) for _ in range(20)),
+        )
+        roles_list.append(role_arn)
+        print("SUCCESS: " + role_arn)
+        
+        # See if I can create a key set automatically in background
+        #if args.create_sessions:
+
+        #    print("CREATING KEYS FOR: " + role_arn)
+        #    new_key_name = f"{resp['AssumedRoleUser']['Arn']}"
+        #    aws.create_session
+        #    session.set_keys(
+        #        key_alias=new_key_name,
+        #        access_key_id=resp['Credentials']['AccessKeyId'],
+        #        secret_access_key=resp['Credentials']['SecretAccessKey'],
+        #        session_token=resp['Credentials']['SessionToken'],
+        #    )
+        #else:
+        #    pass
+
+    except ClientError as error:
+        code = error.response['Error']['Code']
+        if code == 'AccessDenied':
+            print("FAILURE: "+role_arn)
+        else:
+            print('(DEFAULT FAILURE: {}'.format(code))
+
+
+# Main is the first function that is called when this module is executed.
+def main(args, pacu_main):
+    session = pacu_main.get_active_session()
+
+    ###### These can be removed if you are not using the function.
+    args = parser.parse_args(args)
+    print = pacu_main.print
+    fetch_data = pacu_main.fetch_data
+    ######
+
+    accounts = []
+    roles = ["OrganizationAccountAccessRole"]
+    
+    if args.accounts and args.accounts_file:
+        print("You must choose either accounts or the accounts-file argument or neither, you cannot choose both")
+    if args.roles and args.roles_file:
+        print("You must choose either roles or roles-file or neither, you cannot choose both")
+
+    # If user did not specify an account to try brute forcing, then try pre-requisite
+    if not args.accounts and not args.accounts_file:
+        if fetch_data(['Organizations'], 'organizations__enum', '--accounts') is False:
+            return
```

Comment:
Add a print statement before the return keyword that says "Prerequisite module was not run. Exiting...".

Reply:
Fixed